dev-tools/omdb/src/bin/omdb/db.rs

```diff
@@ -2249,6 +2416,1221 @@ async fn cmd_db_network_list_vnics(
     Ok(())
 }
 
+// Copy some types from Steno, because steno uses pub(crate) everywhere. We
+// don't want to change Steno to make the internals public, but these types
+// should be fairly stable.
+
+#[derive(Serialize, Deserialize)]
+struct StenoDag {
+    pub saga_name: String,
+    pub graph: Graph<StenoNode, ()>,
+    pub start_node: NodeIndex,
+    pub end_node: NodeIndex,
+}
+
+impl StenoDag {
+    pub fn get(&self, node_index: NodeIndex) -> Option<&StenoNode> {
+        self.graph.node_weight(node_index)
+    }
+
+    pub fn get_from_saga_node_id(
+        &self,
+        saga_node_id: &nexus_db_model::saga_types::SagaNodeId,
+    ) -> Option<&StenoNode> {
+        self.get(u32::from(saga_node_id.0).into())
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+enum StenoNode {
+    Start { params: Arc<serde_json::Value> },
+    End,
+    Action { name: String, label: String, action_name: String },
+    Constant { name: String, value: Arc<serde_json::Value> },
+    SubsagaStart { saga_name: String, params_node_name: String },
+    SubsagaEnd { name: String },
+}
+
+fn print_sagas(sagas: Vec<Saga>, with_start_params: bool) {
+    if with_start_params {
+        // Using Tabled isn't recommended: the data column could be huge, and Tabled
+        // will print spaces in order to make each column the same size
```

Comment:
I don't quite follow this.  What does this do differently from `Tabled` here?

Reply:
On re-reading, I figured this out.

```suggestion
        // We avoid `Tabled` here because some columns could be very wide and its auto-sizing would cause all rows to be that wide, resulting in unreadable output.
```